### ISSUE TITLE
extend CIDR regex validation

### ISSUE THREAD

**Issue by PatMyron** [aws-cloudformation/cfn-python-lint#1084]
Rules [`W2509`](https://github.com/aws-cloudformation/cfn-python-lint/blob/master/src/cfnlint/rules/parameters/Cidr.py) and [`E2004`](https://github.com/aws-cloudformation/cfn-python-lint/blob/master/src/cfnlint/rules/parameters/CidrAllowedValues.py) check CIDR parameters have `AllowedPattern` and `AllowedValues` are CIDRs respectively

```yaml
Resources:
  ReceiptFilter:
    Type: AWS::SES::ReceiptFilter
    Properties:
      Filter:
        IpFilter:
          Policy: Block
          Cidr: 0.0.0.0/999
```

`Invalid CIDR block: 0.0.0.0/999 (Service: AmazonSimpleEmailService; Status Code: 400; Error Code: InvalidParameterValue; Request ID: )`

* Inline values
* Mappings
* Parameter Defaults
* evaluate AllowedPatterns
